### DISCUSSION TITLE
Is there any doc on how to set up Service Account for reconciling from pipeline?

### DISCUSSION THREAD

**Discussion by MikaelElkiaer** [fluxcd/flux2#2531] [Q&A] [answered]
In Flux1 I had a service account with a Role for `pods/portforward` and all verbs for accessing Flux controller, and ClusterRole `pods` with readonly verbs for obtaining workload status. This was done so that I could run `fluxctl list-workloads` and `fluxctl release` from a GitLab pipeline for synchronous releases.

Is there any doc, or can anyone give me some good pointer, on what is needed for Flux2? Specifically, I would like to do `flux create image policy` and `flux reconcile image update` in my pipeline in order to explicitly set an image and afterwards wait for its deployment. 

This is so that I can lock my staging app to a specific image, run some tests on the service, and then set the same image for the production app.

Thanks!

**Comment by stefanprodan** [comment, marked as answer]
You only need this https://github.com/fluxcd/flux2/blob/main/manifests/rbac/controller.yaml#L18-L20

**Reply by MikaelElkiaer** [reply]
> You only need this https://github.com/fluxcd/flux2/blob/main/manifests/rbac/controller.yaml#L18-L20

Much simpler, thanks.

For anyone else interested in replacing `fluxctl release` for whichever reason, I am combining the flux image access with watch of deployment for a pipeline service account, in order to make a release command as follows:
```
flux create image policy $NAME --image-ref=$NAME --select-alpha=asc --filter-regex="^$(echo $IMAGE_TAG | sed "s,\.,\\\\.,g")\$" \
  && flux reconcile image update $UPDATE_NAME \
  && flux reconcile kustomization $KUSTOMIZE_NAME --with-source \
  && timeout 120s kubectl rollout status deployment/$NAME -w
```

**Reply by stefanprodan** [reply]
For `flux reconcile kustomization` you need: https://github.com/fluxcd/flux2/blob/main/manifests/rbac/controller.yaml#L9-L11

**Reply by MikaelElkiaer** [reply]
> For `flux reconcile kustomization` you need: https://github.com/fluxcd/flux2/blob/main/manifests/rbac/controller.yaml#L9-L11

Thanks for the heads up. Had initially hoped that `flux reconcile image update` had triggered that as well.

**Reply by stefanprodan** [reply]
Flux v2 controllers are independent, so an image update would never trigger a kustomization, you can for example run the image automation controllers on a different cluster as their only purpose is to write back to Git, they don't touch the cluster state.
